devices/webcam/pose_vis/gesture/pose/pose_gesture_vis.py

```diff
@@ -0,0 +1,362 @@
+from pose_vis.display import DisplayHandler
+# from pose_vis.extensions.hands import HandsExtension, HandsConfig, mp_hands
+from pose_vis.extensions.pose import PoseExtension, PoseConfig ,mp_pose
+from pose_vis.performance_utility import PerfUtility
+
+logger = logging.getLogger(__name__)
+
+LANDMARK_DISTANCE = [
+    (mp_pose.PoseLandmark.LEFT_SHOULDER, mp_pose.PoseLandmark.LEFT_WRIST ),
+    # (mp_pose.PoseLandmark.LEFT_SHOULDER, mp_pose.PoseLandmark.LEFT_ELBOW ),
+    # (mp_pose.PoseLandmark.LEFT_SHOULDER, mp_pose.PoseLandmark.LEFT_HIP ),
+    # (mp_pose.PoseLandmark.LEFT_ELBOW, mp_pose.PoseLandmark.LEFT_WRIST ),
+    # ! Maybe add from wrist to thumb/index finger
+
+    (mp_pose.PoseLandmark.RIGHT_SHOULDER, mp_pose.PoseLandmark.RIGHT_WRIST),
+    # (mp_pose.PoseLandmark.RIGHT_SHOULDER, mp_pose.PoseLandmark.RIGHT_ELBOW),
+    # (mp_pose.PoseLandmark.RIGHT_SHOULDER, mp_pose.PoseLandmark.RIGHT_HIP ),
+    # (mp_pose.PoseLandmark.RIGHT_ELBOW, mp_pose.PoseLandmark.RIGHT_WRIST ),
+
+
+
+    (mp_pose.PoseLandmark.LEFT_HIP, mp_pose.PoseLandmark.LEFT_ANKLE ),
+    # (mp_pose.PoseLandmark.LEFT_HIP, mp_pose.PoseLandmark.LEFT_KNEE ),
+    # (mp_pose.PoseLandmark.LEFT_KNEE, mp_pose.PoseLandmark.LEFT_ANKLE ),
+    # ! Maybe distance from ankle to toe
+
+    (mp_pose.PoseLandmark.RIGHT_HIP, mp_pose.PoseLandmark.RIGHT_ANKLE )
+    # (mp_pose.PoseLandmark.RIGHT_HIP, mp_pose.PoseLandmark.RIGHT_KNEE ),
+    # (mp_pose.PoseLandmark.RIGHT_KNEE, mp_pose.PoseLandmark.RIGHT_ANKLE )
+]
+
+LANDMARK_DIRECTION = [
+    # todo --- ADD HERE ---
+]
+
+TORSO_DISTANCE = [
+    (mp_pose.PoseLandmark.LEFT_SHOULDER, mp_pose.PoseLandmark.RIGHT_SHOULDER),
+    (mp_pose.PoseLandmark.LEFT_HIP, mp_pose.PoseLandmark.RIGHT_HIP),
+    (mp_pose.PoseLandmark.LEFT_SHOULDER, mp_pose.PoseLandmark.LEFT_HIP),
+    (mp_pose.PoseLandmark.RIGHT_SHOULDER, mp_pose.PoseLandmark.RIGHT_HIP)
+]
+
+DRAW_DEBUG = False
+
+MAX_DIFFERENCE_VALUE = 3 #! play around with this value
+
+
+#! go over these
+class GV_MODE(Enum):
+    VISUALIZATION = 0,
+    LABEL_INPUT = 1,
+    COLLECTION = 2
+
+#! revisit this class
+@dataclass
+class AnnotationInfo():
+    pose_labels: List[str]
+    pose_bounds: List[List[int]]
+    gesture_data: List[np.ndarray]
+    draw: bool
+
+class PoseGestureVis():
+    
+    sources: List[str | int]
+    resolutions: List[Tuple[int, int, int]]
+    cap_handler: CaptureHandler
+    dis_handler: DisplayHandler
+    perf: PerfUtility
+    annotation_infos: Deque[AnnotationInfo]
+    data_dir: str
+    export_files: List[str]
+    export_format: str
+    running: bool = True
+    mode: GV_MODE = GV_MODE.VISUALIZATION
+    label_name: str = ""
+    label_names: List[str] = []
+    label_data: List[np.ndarray] = []
+    video_writers: List[cv2.VideoWriter]
+
+    # ! Test this
+    def __init__(self, sources: List[str | int], resolutions: List[Tuple[int, int, int]], data_dir: str, export_files: List[str], export_format: str) -> None:
```

Comment:
```List[str | int]``` is not supported in python3.8,
change to ```List[Union[str, int]]``` to support python3.8+
This applies to all instances of ```List[str | int]``` in this PR.